### PR TITLE
:param v2

### PR DESCRIPTION
**Pull request by huboneo**
This PR adds support for the `:param` v2 syntax (explicit returns). In addition, it also cleans up and standardizes the error frame appearance regardless of error origin. Finally it standardizes the appearance of `:help` buttons.

### Todo
- [x] Fix tests once `@neo4j/browser-lambda-parser` is published

### STR
- run `:help param` and follow the instructions for explicit returns

### Screenshots
Setting parameters
Screenshot 2019-08-27 at 15 45 57

Error
Screenshot 2019-08-27 at 14 31 15

`:help` commands
Screenshot 2019-08-27 at 15 43 16

### CODE REVIEW

File: src/shared/modules/commands/helpers/lambdas.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+        return {
+          [parameters.value]: map(extractedRecords, record =>
+            recursivelyTypeGraphItems(record)
+          )
+        }
+      }
+
+      // future proofing
+      if (parameters.type !== ARRAY) return null
```
Comment (huboneo) [inline src/shared/modules/commands/helpers/lambdas.js]: maybe return `{}`

File: package.json [hunk 2]
```diff
@@ -123,6 +123,7 @@
     "xml2js": "^0.4.19"
   },
   "dependencies": {
+    "@neo4j/browser-lambda-parser": "^1.0.2",
```
Comment (oskarhane) [inline package.json]: 🎉

File: src/browser/modules/Help/html/param.html [hunk 3]
```diff
@@ -18,6 +18,12 @@
         <code>:param x => 1</code> and to set it as a float, do
         <code>:param x => 1.0</code>.
       </p>
+      <p>
+        If you need more fine-grained control or advanced Cypher queries, you can use the explicit syntax: <code>x => { ... RETURN 1 as foo }</code>
+        <br/> Explicit returns yield a list of records, matching that of your Cypher query: <code>x => { RETURN 1 as foo }</code> yields <code>$x = [{foo: 1}]</code>
+        <br/> You can pick out individual values from your result using destructuring: <code> [{foo}] => { RETURN 1 as foo }</code> yields <code>$foo = 1</code>
+        <br/> You can also rename destructured params: <code> [{foo: bar}] => { RETURN 1 as foo }</code> yields <code>$bar = 1</code>
+      </p>
```
Comment (oskarhane) [inline src/browser/modules/Help/html/param.html]: 👍

File: src/browser/modules/Stream/auto-exec-button.jsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  line-height: 18px;
+  padding: 0 4px;
+  color: #428bca;
+  cursor: pointer;
+  text-decoration: none;
+  background-color: #f8f8f8;
+  outline: transparent;
+`
+
+function AutoExecButtonComponent ({ bus, cmd, cmdChar }) {
```
Comment (oskarhane) [inline src/browser/modules/Stream/auto-exec-button.jsx]: Could you add tests for this component, to make sure what we expect is send over the bus, and that the rendered output is what's expected?
We probably have it somewhat covered in the e2e tests, but it's faster feedback if it breaks already here if we introduce a regression in the future.

**Comment by huboneo** [comment]
@oskarhane thank you for your feedback! I have addressed your comments. For the AutoExecButton tests I had to export a non-bus version as I could not figure out how to properly mock `react-suber` Please let me know if you have a better way

**Pull request merged**